test/e2e/istio_test.go

```diff
@@ -313,39 +323,58 @@ func TestIstioWithKongIngressGateway(t *testing.T) {
 // -----------------------------------------------------------------------------
 
 // verifyStatusForUrl is a helper function which given a URL and a status code performs
-// a GET and verifies the status code returned.
-func verifyStatusForUrl(t *testing.T, getURL string, statusCode int) {
+// a GET and verifies the status code returning an error if the result isn't as expected.
+func verifyStatusForUrl(t *testing.T, getURL string, statusCode int) error {
 	resp, err := httpc.Get(getURL)
-	require.NoError(t, err)
+	if err != nil {
+		return err
+	}
 	defer resp.Body.Close()
-	require.Equal(t, statusCode, resp.StatusCode)
+	if resp.StatusCode != statusCode {
+		return fmt.Errorf("expected status code %d got %d", statusCode, resp.StatusCode)
+	}
+	return nil
 }
 
 // getKialiWorkloadHealth produces the health metrics of a workload given the namespace and name of that workload.
-func getKialiWorkloadHealth(t *testing.T, kialiAPIUrl string, namespace, workloadName string) workloadHealth {
+func getKialiWorkloadHealth(t *testing.T, kialiAPIUrl string, namespace, workloadName string) (workloadHealth, error) {
 	// generate the URL for the namespace health metrics
 	kialiHealthURL := fmt.Sprintf("%s/namespaces/%s/health", kialiAPIUrl, namespace)
 	req, err := http.NewRequest("GET", kialiHealthURL, nil)
-	require.NoError(t, err)
+	if err != nil {
+		return workloadHealth{}, err
```

Comment:
nit: maybe change the return type to return a pointer instead of a struct?

Reply:
Sure thing, done in 8e50f69b7a4efda882212a639993cded7392c8d6